RecoVertex/Configuration/python/RecoVertex_cff.py

```diff
@@ -34,18 +34,19 @@
 
 #timing
 from RecoVertex.PrimaryVertexProducer.TkClusParameters_cff import DA2D_vectParameters
-unsortedOfflinePrimaryVertices1D = unsortedOfflinePrimaryVertices.clone()
-unsortedOfflinePrimaryVertices1D.TkFilterParameters.minPt = cms.double(0.7)
-offlinePrimaryVertices1D=sortedPrimaryVertices.clone(vertices="unsortedOfflinePrimaryVertices1D", particles="trackRefsForJetsBeforeSorting")
-offlinePrimaryVertices1DWithBS=sortedPrimaryVertices.clone(vertices="unsortedOfflinePrimaryVertices1D:WithBS", particles="trackRefsForJetsBeforeSorting")
 DA2D_vectParameters.TkDAClusParameters.verbose = cms.untracked.bool(False)
 unsortedOfflinePrimaryVertices4D = unsortedOfflinePrimaryVertices.clone( verbose = cms.untracked.bool(False),
                                                                          TkClusParameters = DA2D_vectParameters )
-unsortedOfflinePrimaryVertices4D.TkFilterParameters.minPt = cms.double(0.7)
+unsortedOfflinePrimaryVertices4D.TkFilterParameters.minPt = cms.double(0.0)
 unsortedOfflinePrimaryVertices4D.TrackTimesLabel = cms.InputTag("trackTimeValueMapProducer:generalTracksConfigurableFlatResolutionModel")
 unsortedOfflinePrimaryVertices4D.TrackTimeResosLabel = cms.InputTag("trackTimeValueMapProducer:generalTracksConfigurableFlatResolutionModelResolution")
-offlinePrimaryVertices4D=sortedPrimaryVertices.clone(vertices="unsortedOfflinePrimaryVertices4D", particles="trackRefsForJetsBeforeSorting")
-offlinePrimaryVertices4DWithBS=sortedPrimaryVertices.clone(vertices="unsortedOfflinePrimaryVertices4D:WithBS", particles="trackRefsForJetsBeforeSorting")
+offlinePrimaryVertices4D=sortedPrimaryVertices.clone(vertices="unsortedOfflinePrimaryVertices4D", particles="trackRefsForJetsBeforeSorting4D", trackTimeTag=cms.InputTag("trackTimeValueMapProducer","generalTracksConfigurableFlatResolutionModel"),trackTimeResoTag=cms.InputTag("trackTimeValueMapProducer","generalTracksConfigurableFlatResolutionModelResolution"),assignment=dict(useTiming=True))
```

Comment:
given changes in inputs: how do the PV validation plots look like for 4D vertices compared to the last baseline?
(with phase-2 PU)

Reply:
You mean compared to what things look like with https://github.com/cms-sw/cmssw/pull/19935?

Haven't run the PV validation plots, but in other private tests verified that these changes (actually the most important is removing the pt>0.7 GeV cut) have the expected effect in terms of increasing the number of primary vertices (from being below the number for the standard 3d vertices, to slightly above, as expected from the reduction in merging.)

Reply:
> You mean compared to what things look like with #19935?

yes.

Simply an increase in NPV is not a clear enough target. Was there cost-benefit analysis?

As an example, in the update of current PV in 2017 #18330 there were clear reasons to increase PT threshold at least in some regions based on analysis of associated tracks.

Reply:
BTW, did the timing stay under control? What is it for PU200 before and after?

Reply:
So the original reason for the pt cut was because older versions of the time association code did not deal with the issue of the timing layer acceptance, and this was being imposed at the level of the 4d vertexing inputs.  Now the time association properly imposes the acceptance, and there is proper handling in the vertexing of tracks with vs without timing information (via the fact that the timeError is set to negative for tracks without time information).  So the original reason for the pt cut went away, and we just decided to synchronize with what is used for the standard 3d vertexing.

Reply:
And sorry, I don't see any non-zero pt cut in https://github.com/cms-sw/cmssw/pull/18330

Reply:
you are right.
I have confused the actual changes with discussion about pt-dependent track selection related to a plot like https://github.com/cms-sw/cmssw/pull/18330#issuecomment-297004335

Reply:
Can check the timing in any case.  Also are there instructions somewhere how to easily produce comparison plots like the ones you posted here: (since this is I guess what you would like to see)
https://github.com/cms-sw/cmssw/pull/19935#issuecomment-328767866

Reply:
the same scripts that are run by jenkins (in alternative-comparisons or validateJR) can be made following 
https://twiki.cern.ch/twiki/bin/viewauth/CMS/RecoIntegration#Compare_outputs_DQM_style

Using RelMon is another option.
Individual histograms could also be picked directly from the harvested DQM files,
but that becomes tedious for n>3-5.

Reply:
Do I need to run also the tests for the reference myself, are are the outputs from the latest IB accessible somewhere?

Reply:
it would be most consistent to produce both

Reply:
@slava77 how many events did you produce for the plots in https://github.com/cms-sw/cmssw/pull/18330#issuecomment-297004335 ?  I assume this was 0 pileup?

Reply:
that plot was from @werdmann 
I think that was with PU, but my memory is not so certain

Reply:
While trying to get these validation plots, let me also say that these changes have also been used and validated to some level as part of the recent timing studies (e.g some of the comparisons between different versions made here, with the latest version containing all of these changes https://indico.cern.ch/event/665516/contributions/2718730/attachments/1522842/2379773/13-09-17_upsg_slides.pdf )